### PR TITLE
ES6

### PR DESCRIPTION
**Pull request by ocke**
Split up the work started in https://github.com/zendesk/zendesk_app_framework_sdk/pull/71

ES6 code!

**review by (only 3) commits**

### es6
es6

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -22,33 +19,30 @@ var ZAFClient = {}
...
+    }
+
+    return client
   }
-
-  return client
 }
 
-module.exports = ZAFClient
+window.ZAFClient = ZAFClient
```
[the remark this answers is not in the document]
Comment (ocke) [reply lib/index.js]: So others can use it?

File: webpack.config.js [hunk 2]
```diff
@@ -11,13 +11,15 @@ module.exports = function (env = {}) {
       'zaf_sdk': [
         'native-promise-only',
         './lib/index.js'
-      ],
```
Comment (ocke) [inline webpack.config.js]: I moved this to the `if production` condition, because for tests this was taking forever.

File: spec/index_spec.js [hunk 3]
```diff
@@ -1,50 +1,18 @@
...
-      describe('when a callback is passed', function () {
-        var callback = function () { return 'abcxyz' }
-        callback.bind = function () { return callback }
-
-        beforeEach(function () {
-          sandbox.spy(Client.prototype, 'on')
-          ZAFClient.init(callback)
-        })
-
-        it('binds the callback to app.registered', function () {
```
Comment (danielbreves) [inline spec/index_spec.js]: Where did these tests go?

Comment (ocke) [reply spec/index_spec.js]: they are back! 🎉

File: spec/utils_spec.js [hunk 4]
```diff
@@ -1,12 +1,12 @@
 /* eslint-env mocha */
-/* global expect */
-describe('Utils', function () {
-  var Promise = window.Promise || require('native-promise-only')
```
Comment (hoangtrvu) [inline spec/utils_spec.js]: Do we still need to import Promise here?

File: webpack.config.js [hunk 5]
```diff
@@ -45,5 +47,24 @@ module.exports = function (env = {}) {
...
 
+  if (env.production) {
+    config.entry['zaf_sdk.min'] = [
+      'native-promise-only',
+      './lib/index.js'
+    ]
+
+    config.module.rules.push({
+      test: /\.js$/,
+      use: { loader: 'babel-loader', options: { plugins: [], presets: ['babel-preset-env'] } }
```
Comment (hoangtrvu) [inline webpack.config.js]: Should we use babel for development as well, for example, to test with various browser/os?

File: spec/utils_spec.js (hunk 4 above)
Comment (ocke) [reply spec/utils_spec.js]: I did a bit of digging, but that line is very old from when we were running in phantom, not in chrome. We only run in chrome now, so Promise will always be available.

File: lib/client.js [hunk 6]
```diff
@@ -240,7 +241,7 @@ function isOriginValid (origin) {
       /^https:\/\/dashboard\.zopim\.org$/
     ]
 
-    for (var i = 0; i < WHITELISTED_ORIGINS.length; i++) {
+    for (let i = 0; i < WHITELISTED_ORIGINS.length; i++) {
```
[the remark this answers is not in the document]
Comment (ocke) [reply lib/client.js]: as discussed.

**Pull request merged**